helper/genericdrive.h

```diff
@@ -17,30 +17,25 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-#ifndef RESTOREJOB_H
-#define RESTOREJOB_H
+#ifndef GENERICDRIVE_H
+#define GENERICDRIVE_H
 
 #include <QObject>
-#include <QProcess>
-#include <QTextStream>
+#include <QString>
 
-class RestoreJob : public QObject
-{
+class GenericDrive : public QObject {
     Q_OBJECT
 public:
-    explicit RestoreJob(const QString &where);
-
-signals:
-
-private slots:
-    void work();
-
-private:
-    QTextStream out { stdout };
-    QTextStream err { stderr };
-
-    QProcess m_diskpart;
-    int m_where;
+    virtual void init() = 0;
```

Comment:
Another sign that I should use composition here.

Reply:
Or actually make use of inheritance and polymorphism.

Reply:
It seems to be that the reason why I added these unused virtual methods `init` and `wipe` is so that the linker would throw an error if I forgot to implement them. The others are required because they're referred to in `genercdrive.cpp`.